hooks/pre/17-memory_check.rb

```diff
@@ -0,0 +1,20 @@
+# define our error message method to write to stderr and exit
+
+def error_exit(message, code)
+  $stderr.puts message
+  exit code
+end
+
+# call grep and awk to store total ram and define min ram
+total_ram = %x( grep MemTotal /proc/meminfo | awk '{print $2}' ).to_i
+min_ram = 8388608
+       
+# call mem_check if flag is called
+if app_value(:disable_system_checks) then
+logger.warn 'Skipping system checks.'
+else
+ if min_ram > total_ram
+logger.error 'This system has less than 8 GB of total memory. Please have at least 8 GB of total ram free before running the installer.'
+kafo.class.exit 1
+ end
```

Comment:
Please fix indentation, all ruby code needs to follow [our style guide](http://www.katello.org/developers/style/ruby.html)